### PR TITLE
Fix not returning error when creating an user with invalid e-mail

### PR DESCRIPTION
**Pull request by netoax**
**Describe what this PR introduces:**

If you try to create a new user with an invalid e-mail such as "quarentine.com", the user's service returns a bad request error that informs the content type isn't supported. However, the babeltower isn't handling this error, so the response comes with a successful status code and a JSON object with an empty token. In order to solve it, this PR adds error handling and use this moment to refactor the code in order to improve readability and maintainability.

**What kind of change does this PR introduce?** (check at least one)

- [X] Bug fix
- [ ] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce any breaking changes?** (check one)

- [ ] Yes
- [X] No

**The PR fulfills these requirements:**

- [ ] Related issues are referenced in the PR (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [X] All tests are passing
- [X] New/updated tests are included

**Testing environment:**

- Operating System/Platform: macOS Darwin 18.0.0
- Go version: 1.14

### CODE REVIEW

**Review by ramonhpr — CHANGES_REQUESTED** [review]
The patch is good but what do you think to create a test to validate this error? It should fail before your changes and pass after them

**Comment by netoax** [comment]
> The patch is good but what do you think to create a test to validate this error? It should fail before your changes and pass after them

Thanks for the feedback. Done.

File: pkg/mocks/user_proxy.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+	"github.com/CESARBR/knot-babeltower/pkg/user/entities"
+	"github.com/stretchr/testify/mock"
+)
+
+// FakeUserProxy represents a mocking type for the user's proxy service
+type FakeUserProxy struct {
+	mock.Mock
+	Token string
+	Err   error
+	// Thing     *entities.Thing
```
Comment (lcbm) [inline pkg/mocks/user_proxy.go]: Oopss :grimacing: 

```suggestion
```

**Review by lcbm — APPROVED** [review]
Approving this in advance, but please remove the commented line :joy:

File: pkg/mocks/user_proxy.go (hunk 1 above)
Comment (netoax) [reply pkg/mocks/user_proxy.go]: OMG, haha. Tomorrow morning I'll remove it.

Comment (netoax) [reply pkg/mocks/user_proxy.go]: Done.

**Pull request merged**